server/src/sentence-model/SentenceCollection.ts

```diff
@@ -353,10 +353,31 @@ export class SentenceCollection implements vscode.TextDocument {
             this.sentences[start-1].next = null;
           //
           removed.forEach((sent) => sent.dispose());
+          if (reparsed.length > 10){
+            console.log("NOTICE: some editing change lead us to reparse until EOF.")
+          }
           return {removed: removed, added: reparsed, endOfSentences: true};
-        } if(idx >= minCount && start+idx < this.sentences.length && currentOffset+sent.text.length === this.sentences[start+idx].getDocumentEndOffset() && sent.text === this.sentences[start+idx].getText()) {
+        }
+        
+        var fixByLocalGlueing : undefined | number  =  undefined;
+
+        if(idx >= minCount && start+idx < this.sentences.length
+            && currentOffset+sent.text.length === this.sentences[start+idx].getDocumentEndOffset()
```

Comment:
Does anyone understand why we use getDocumentEndOffset here?

Reply:
Why is the offset of the already parsed, old sentences comparable with our current position after parsing new things?